force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +113,19 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
+        Integer result = 0;
+        List<String> vowelList = new List<String> {'a', 'e', 'i', 'o', 'u'};
+        
+        for (Integer i = 0; i < str.length(); i++) {
+            String oneChar = str.substring(i, i+1);
+            if (vowelList.contains(oneChar)) {
+                result++;
+            }
             // get individual characters from the string - hint use substring or split
             // check if the character is a vowel
             // if it is a vowel, increment the result Integer
         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        return result; // Replace null with the variable you used to store the result
```

Comment:
This is a little more of a computer science topic. But when you are checking for membership, a Set is the data structure you want to use. That will give you a much more performant solution.  Due to the way a Set uses a hashing algorithm 

```
  public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        for (String character : str.split(''))  {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
 ```